### PR TITLE
[AIRFLOW-356] [AIRFLOW-355] [AIRFLOW-354] Replace nobr, enable DAG only exists locally message, change edit DAG icon

### PR DESCRIPTION
**Pull request by ldct**
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:
- [https://issues.apache.org/jira/browse/AIRFLOW-356](https://issues.apache.org/jira/browse/AIRFLOW-356)
- [https://issues.apache.org/jira/browse/AIRFLOW-355](https://issues.apache.org/jira/browse/AIRFLOW-355)
- [https://issues.apache.org/jira/browse/AIRFLOW-354](https://issues.apache.org/jira/browse/AIRFLOW-354)
#### Changes:
- Replace `<nobr>` with `flexbox`
- "This DAG seems to be existing only locally" now shows up
- Change edit DAG icon from info to edit
- Rename `dttm` variable to `file_last_changed_on_disk`
- Rename `dags` variable to `webserver_dags` 
- Adds a comment clarifying what `self.file_last_changed` is
- Clarifies what the `dag.last_expired` column represents
- Refactors some previously very nested logic in `views.py` and adds comments
- Properly indents `dags.html` and adds comments to it
#### User-visible changes:
- Edit DAG icon changed
- Home page now sort of responsive, no longer fixed width
- User will occasionally see "This DAG seems to be existing only locally" message
#### Testing Done:
- Verify that edit dag button is now an edit icon and click on it
- Resized home page, check that last column does not wrap

image

### CODE REVIEW

File: scripts/ci/requirements.txt [hunk 1]
```diff
@@ -14,7 +14,7 @@ flake8
 flask
 flask-admin
 flask-cache
-flask-login
+flask-login==0.2.11
```
Comment (aoen) [inline scripts/ci/requirements.txt]: Can you rebase off of master to make this change go away (since IIRC it was in your other PR)?

File: airflow/www/views.py [hunk 2]
```diff
@@ -1654,30 +1655,37 @@ def index(self):
...
+            # only show dags owned by @current_user.user.username
+            webserver_dags = {
+                dag.dag_id: dag
+                for dag in webserver_dags
+                if dag.owner == current_user.user.username
+            }
         else:
-            dags = {dag.dag_id: dag for dag in dags if not dag.parent_dag}
-        all_dag_ids = sorted(set(orm_dags.keys()) | set(dags.keys()))
+            webserver_dags = {
```
Comment (aoen) [inline airflow/www/views.py:1678]: you can pull this out of and before the if statement, also this should be webserver_dag_ids (otherwise it's confusing to use the same variable to store two different types of data).

File: airflow/www/views.py [hunk 3]
```diff
@@ -1654,30 +1655,37 @@ def index(self):
...
-                    if (
-                        dag.owner == current_user.user.username and (not dag.parent_dag)
-                    )
-                }
+
+        # read webserver_dags from local dagbag
+
+        webserver_dags = dagbag.dags.values()
+        # remove subdags
+        webserver_dags = [dag for dag in webserver_dags if not dag.parent_dag]
```
Comment (aoen) [inline airflow/www/views.py]: Combine the two lines above, and I would also name it unfiltered_webserver_dags to differentiate it from the final state that the webserver_dags variable tasks on.

File: airflow/models.py [hunk 4]
```diff
@@ -2441,8 +2443,7 @@ class DagModel(Base):
     last_scheduler_run = Column(DateTime)
     # Last time this DAG was pickled
     last_pickled = Column(DateTime)
-    # When the DAG received a refreshed signal last, used to know when
-    # we need to force refresh
+    # The last time someone clicked on refresh
```
Comment (aoen) [inline airflow/models.py]: Nit: # When the DAG last received a refresh signal (e.g. the DAG's "refresh" was clicked in the web UI)

File: airflow/models.py [hunk 5]
```diff
@@ -201,10 +202,11 @@ def get_dag(self, dag_id):
                     dag.last_loaded < orm_dag.last_expired
                 )
         ):
-            # Reprocessing source file
+            # Then reprocess source file
             found_dags = self.process_file(
                 filepath=orm_dag.fileloc, only_if_updated=False)
 
+            # If the source file now does not export `dag_id`, delete it from self.dags
```
Comment (aoen) [inline airflow/models.py]: Nit: remove "now"

File: airflow/models.py [hunk 6]
```diff
@@ -201,10 +202,11 @@ def get_dag(self, dag_id):
                     dag.last_loaded < orm_dag.last_expired
                 )
         ):
-            # Reprocessing source file
+            # Then reprocess source file
```
Comment (aoen) [inline airflow/models.py]: Nit: Why this change? I kind of prefer the old version.

**Comment by aoen** [comment]
What will the text look like without the <nobr> tag if we don't have flexbox set up yet?
Can you stage the webserver on Airbnb infra to be safe?

File: airflow/models.py (hunk 6 above)
Comment (ldct) [inline airflow/models.py]: I changed it to match https://github.com/apache/incubator-airflow/pull/1678/files#diff-a32a363fa616685db3bfefba947535b2L195 ("If the root_dag_id is absent or expired...then reprocess source file")

File: airflow/models.py (hunk 5 above)
Comment (ldct) [inline airflow/models.py]: I included "now" because the source file must have previously exported `dag_id` for the `del` to be run (I think)

File: airflow/models.py (hunk 6 above)
Comment (aoen) [inline airflow/models.py]: Ah it's kind of hard to understand since the comment is split up, I still prefer the older version (you don't need to read the comment above to understand what this comment does).

It's a nit so feel free to ignore.

File: airflow/models.py (hunk 5 above)
Comment (aoen) [inline airflow/models.py]: Ah maybe "If the source file no longer exports `dag_id`" then?

File: airflow/www/views.py (hunk 2 above)
Comment (ldct) [inline airflow/www/views.py:1678]: it shouldn't be called `webserver_dag_ids`, it's a dictionary of `dag_id: dag`

the code could be converted to do the filtering and the list -> dict conversion separately

**Comment by aoen** [comment]
LGTM once staged (just making sure the front-page loads).
